src/commands/server/update.ts

```diff
@@ -110,40 +120,114 @@ export default class Update extends Command {
-          || cheCluster.spec.server.cheImageTag
-          || cheCluster.spec.server.devfileRegistryImage
-          || cheCluster.spec.database.postgresImage
-          || cheCluster.spec.server.pluginRegistryImage
-          || cheCluster.spec.auth.identityProviderImage) {
-          cli.warn(`In order to update Eclipse Che the images defined in the '${cheCluster.metadata.name}'
-            Custom Resource of the namespace '${flags.chenamespace}' will be cleaned up:`)
-          cheCluster.spec.server.cheImageTag && cli.warn(`Eclipse Che server image tag [${cheCluster.spec.server.cheImageTag}]`)
-          cheCluster.spec.server.cheImage && cli.warn(`Eclipse Che server [${cheCluster.spec.server.cheImage}]`)
-          cheCluster.spec.database.postgresImage && cli.warn(`Database [${cheCluster.spec.database.postgresImage}]`)
-          cheCluster.spec.server.devfileRegistryImage && cli.warn(`Devfile registry [${cheCluster.spec.server.devfileRegistryImage}]`)
-          cheCluster.spec.server.pluginRegistryImage && cli.warn(`Plugin registry [${cheCluster.spec.server.pluginRegistryImage}]`)
-          cheCluster.spec.auth.identityProviderImage && cli.warn(`Identity provider [${cheCluster.spec.auth.identityProviderImage}]`)
+      // the same version is already installed
+      if (newOperatorImage === existedOperatorImage) {
+        if (projectName === 'chectl' && latestChectlVersion) {
+          // suggest update chectl first
+          if (currentChectlVersion !== latestChectlVersion) {
+            cli.warn(`It is not possible to update Eclipse Che to a newer version
+using the current '${currentChectlVersion}' version of chectl. Please, update 'chectl'
+to a newer version '${latestChectlVersion}' with the command 'chectl update ${chectlChannel}'
+and then try again.`)
+          } else if (!flags[CHE_OPERATOR_CR_PATCH_YAML_KEY]) {
+            // same version, no patch then nothing to update
+            cli.info('Eclipse Che is already up to date.')
+            this.exit(0)
+          }
+        } else {
+          // unknown project, no patch file then suggest to update
+          if (!flags[CHE_OPERATOR_CR_PATCH_YAML_KEY]) {
+            cli.warn(`It is not possible to update Eclipse Che to a newer version
+using the current '${currentChectlVersion}' version of '${getCurrentChectlName()}'.
+Please, update '${getCurrentChectlName()}' and then try again.`)
+            this.exit(0)
+          }
         }
+        // custom operator image is used
+      } else if (newOperatorImage !== DEFAULT_CHE_OPERATOR_IMAGE) {
+        cli.warn(`Eclipse Che operator deployment will be updated with the provided image,
+but other Eclipse Che components will be updated to the ${defaultOperatorImageTag} version.
+Consider removing '--che-operator-image' to update Eclipse Che operator to the same version.`)
+      }
+
+      if (!flags.yes && !await cli.confirm('If you want to continue - press Y')) {
+        this.exit(0)
```

Comment:
I think it would be nice to print something into output, like `Update cancelled by user`

Reply:
Agree